src/store/modules/SearchQuery.js

```diff
@@ -1,127 +1,42 @@
-const initialQuery = {
+import filters from '@/store/modules/filterSubModule';
+
+const defaultQuery = {
   searchPhrase: '',
-  type: 'any',
   page: 1,
-  filters: {
-    lastSeen: '[ now/d-30d TO *]',
-    size: null,
-  },
 };
 
-/**
- * Return query params for consumption by $router.push({query})
- * Inverse of setQueryFromParams()
- * @param state
- * @returns {{q: (*|string), last_seen: Function, size: Function, page, type}}
- */
-function stateToQueryParams(state) {
-  return {
-    q: state.searchPhrase,
-    last_seen: state.filters.lastSeen,
-    size: state.filters.size,
-    type: state.type,
-    page: state.page,
-  };
-}
-
-function getLastSeenFilters(lastSeen) {
-  if (lastSeen) {
-    return [`last-seen:${lastSeen}`];
-  }
-
-  return [];
-}
-
-function getSizeFilters(size) {
-  if (size) {
-    if (Array.isArray(size)) {
-      return size.map((value) => `size:${value}`);
-    }
-
-    // Singular value
-    return [`size:${size}`];
-  }
-
-  return [];
-}
-
-function getFilters(filters) {
-  const lastSeenFilters = getLastSeenFilters(filters.lastSeen);
-  const sizeFilters = getSizeFilters(filters.size);
-  return [...lastSeenFilters, ...sizeFilters];
-}
-
 /**
  * flatten query and filters into a string, but not the page or the filetype
  * @param state
  * @returns {string}
  */
-function apiQueryString(state) {
-  return [state.searchPhrase, ...getFilters(state.filters)].join(' ');
+function apiQueryString(state, getters) {
+  return [state.searchPhrase, ...getters['filters/mapFiltersToApi']].join(' ');
 }
 
 const getters = {
-  stateToQueryParams,
   apiQueryString,
 };
 
 const mutations = {
-  // Mutations relating to query composition
-  setRouteParams(state, params) {
-    // Inverse of getters.queryParams
-    state.searchPhrase = params.q || initialQuery.searchPhrase;
-    state.type = params.type || initialQuery.type;
-    state.page = Number(params.page) || initialQuery.page;
-    state.filters = {
-      lastSeen: params.last_seen || initialQuery.filters.lastSeen,
-      size: params.size || initialQuery.filters.size,
-    };
-  },
-  setPage(state, page) {
-    state.page = page;
-  },
-  incrementPage(state) {
-    state.page += 1;
-  },
-  decrementPage(state) {
-    if (state.page >= 1) {
-      // Never decrease below 0
-      state.page -= 1;
-    }
-  },
-  setLastSeenFilter(state, lastSeen) {
-    state.filters.lastSeen = lastSeen;
-  },
-  setSizeFilter(state, size) {
-    state.filters.size = size;
+  setRouteParams(state, routeParams) {
+    // map query parameters to state
+    state.searchPhrase = routeParams.q || defaultQuery.searchPhrase;
+    state.page = Number(routeParams.page) || defaultQuery.page;
+    Object.entries(state.filters).forEach(([slug, filter]) => {
+      filter.select(routeParams[slug]);
+    });
```

Comment:
It would be great to have a little documented example of some filter and search queries here in a comment, that would greatly help with future maintenance as well as provide a basis for implementing tests.